neuroanalysis/data/dataset.py

```diff
@@ -33,10 +33,18 @@ class Container(object):
     """
     def __init__(self):
         self._meta = OrderedDict()
+        self._key = None
+        self._parent = None
         
     @property
     def parent(self):
-        return None
+        return self._parent
+
+    @property
+    def top_parent(self):
+        while self.parent is not None:
```

Comment:
My goal was to be able to call top_parent from any level of Container subclass and be able to get the dataset (in this case). The only way I could think of to do that was to go up in parents until there wasn't a parent. Is there a better way?

Reply:
I get that you have a recursive method call here, but the while-loop immediately returns--it can only run the inner block at most once, which means it's the same as an if-then:
```
if self.parent is not None:
    return self.parent.top_parent
else:
    return self
````

Reply:
Ah, thank you! I see now.